sqlglot/dialects/postgres.py

```diff
@@ -56,12 +56,15 @@ def func(self: Postgres.Generator, expression: DATE_ADD_OR_SUB) -> str:
         this = self.sql(expression, "this")
         unit = expression.args.get("unit")
 
-        expression = self._simplify_unless_literal(expression.expression)
-        if not isinstance(expression, exp.Literal):
+        e = self._simplify_unless_literal(expression.expression)
+        if isinstance(e, exp.Literal):
+            e.args["is_string"] = True
+        elif e.is_number:
+            e = exp.Literal.string(e.to_py())
```

Comment:
Is the order in the branches correct?

 If `e.is_number` is `True` then `e` is an `exp.Literal(is_string=False)`, I think the `elif` branch won't be hit

Reply:
> If `e.is_number` is `True` then `e` is an `exp.Literal(is_string=False)`

This isn't true, `is_number` now handles `Neg` expressions as well. This condition is supposed to handle `Neg`s.

Reply:
Sg, missed the new `is_number` definition